src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -0,0 +1,701 @@
+        IE_THROW() << "TileEmitter constructor got inconsistent arguments. Check num_inputs + num_outputs == io_dims.size()";
+}
+
+void TileEmitter::emit_code(const std::vector<size_t> &in,
+                            const std::vector<size_t> &out,
+                            const std::vector<size_t> &pool,
+                            const std::vector<size_t> &gpr) const {
+    validate_arguments(in, out, pool, gpr);
+    emit_impl(in, out, pool, gpr, nullptr);
+}
+
+void TileEmitter::validate_arguments(const std::vector<size_t> &in,
+                                     const std::vector<size_t> &out,
+                                     const std::vector<size_t> &pool,
+                                     const std::vector<size_t> &gpr) const {
+    if (in.size() != 1)
+        IE_THROW() << "TileEmitter got invalid number of inputs. Expected 1, got " << in.size();
+    if (out.size() != io_dims.size())
+        IE_THROW() << "TileEmitter got invalid number of outputs. Expected " << io_dims.size() << " , got " << out.size();
+}
+
+void TileEmitter::emit_body(const std::vector<size_t>& vec_pool, const std::vector<size_t>& gpr_pool) const {
+    for (auto& code : body)
+        code.first->emit_code(code.second.first, code.second.second, vec_pool, gpr_pool);
+}
+
+void TileEmitter::emit_ptr_increments(const std::vector<Reg64>& data_ptr_regs) const {
+    for (size_t i = 0; i < num_inputs + num_outputs; i++) {
+        // those with dims == 1 will be broadcasted, hence don't require increment
+        if (io_dims[i] != 1)
+            h->add(data_ptr_regs[i], increment * io_data_size[i]);
+    }
+}
+
+void TileEmitter::emit_impl(const std::vector<size_t>& in,
+                            const std::vector<size_t>& out,
+                            const std::vector<size_t>& vec_pool,
+                            const std::vector<size_t>& gpr_pool,
+                            const ov::intel_cpu::emitter_context *emit_context) const {
+    Reg64 work_amount = Reg64(static_cast<int>(in[0]));
+    std::vector<Reg64> data_ptr_regs;
+    transform_idxs_to_regs(out, data_ptr_regs);
+    Label for_body;
+    // Note that:
+    // * Work amount must be set by TileScheduler that executes Tiles
+    // * TileScheduler executes Tile only if it has to perform >= 1 iterations
+    h->L(for_body);
+    emit_body(vec_pool, gpr_pool);
+    emit_ptr_increments(data_ptr_regs);
+    h->sub(work_amount, increment);
+    h->cmp(work_amount, increment);
+    h->jge(for_body, CodeGenerator::T_NEAR);
+}
+
+BroadcastMoveEmitter::BroadcastMoveEmitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl::cpu::x64::cpu_isa_t isa,
+                                           const std::shared_ptr<ov::Node>& n) : jit_emitter(h, isa, n) {
+    if (n->get_input_shape(0).empty())
+        use_broadcast = true;
+    else if (*n->get_input_shape(0).rbegin() != *n->get_output_shape(0).rbegin())
+        use_broadcast = true;
+    else
+        use_broadcast = false;
```

Comment:
As I remember, BroadcastMove is inserted when we should use broadcasting. For example,
Inputs: `[2,16,1] [2,16,10] ` -> we insert BroadcastMove after first input

Also we should insert `fake` BroadcastMove in the following case:
Inputs: `[2,1,10] [2,16,10] ` -> we insert BroadcastMove after first input as well. But in the reality this emitter just move data from src to dst vmm because we have BroadcastMove only for broadcasting first dimension (`1->16`)

Reply:
So you are saying that we always generate useless instruction in case of outer tile broadcasting? Nice)
I'll have to take a closer look on this